### PR TITLE
Fix POST request when submit

### PR DESCRIPTION
**Pull request by fcheval**

### CODE REVIEW

File: personalitytest/static/js/src/personalitytest-view.js [hunk 1]
```diff
@@ -97,8 +97,8 @@ function PersonalityTestXBlockStudent(runtime, element) {
     getScore();
     initDisplay();
 
-    $('.personality-test-form').on('click', '.save-button', function () {
-        //e.preventDefault();
+    $('.personality-test-form').on('click', '.save-button', function (e) {
+        e.preventDefault();
         // addGroupForm.off('submit');
```
Comment (tusbar) [inline personalitytest/static/js/src/personalitytest-view.js]: remove this comment too?

Comment (fcheval) [inline personalitytest/static/js/src/personalitytest-view.js]: Checked !

Comment (tusbar) [inline personalitytest/static/js/src/personalitytest-view.js]: checked?

Comment (fcheval) [inline personalitytest/static/js/src/personalitytest-view.js]: ca veut dire que c'est fait, en anglais

Comment (tusbar) [inline personalitytest/static/js/src/personalitytest-view.js]: c’est “done” alors.

File: personalitytest/static/css/personalitytest.css [hunk 2]
```diff
@@ -1,12 +0,0 @@
...
-}
-
-.answer-cell {
-  vertical-align: middle;
-  padding-bottom: 20px;
-}
-
-.full-result-table {
-  width: 50%;
-}
```
Comment (tusbar) [inline personalitytest/static/css/personalitytest.css:12]: if you remove the CSS file you need to stop importing it [here](https://github.com/IONISx/xblock-personality-test/blob/master/personalitytest/personalitytestblock.py#L141) then.

File: personalitytest/static/js/src/personalitytest-view.js [hunk 3]
```diff
@@ -7,10 +7,13 @@ function PersonalityTestXBlockStudent(runtime, element) {
...
+                var list = document.createElement('ol');
 
                 questions.forEach(function (question) {
-                    var tblRow = tblBody.insertRow();
+                    var questionInList = document.createElement('li');
+                    var spanQuestion = document.createElement('div');
+
                     var select = document.createElement('select');
                     var opt = document.createElement('option');
                     opt.text = '-';
```
Comment (tusbar) [inline personalitytest/static/js/src/personalitytest-view.js]: this should be just an empty string.

File: personalitytest/static/js/src/personalitytest-view.js [hunk 4]
```diff
@@ -24,18 +27,22 @@ function PersonalityTestXBlockStudent(runtime, element) {
...
+                    list.appendChild(questionInList);
 
-                    $('.personality-test-form-table', element).append(tblBody);
+                    mainDiv.append(list);
                 });
                 var submit = $('<div class="action panel-body">');
                 var button = $('<button class="save-button" type="submit">').text('Submit');
                 var errorSpan = $('<span class="error-span"></span>');
                 var submitDiv = $('<div class="submitDiv"></div>');
                 errorSpan.addClass('errorSpan');
```
Comment (tusbar) [inline personalitytest/static/js/src/personalitytest-view.js]: why add `errorSpan` here when you already added `error-span` 2 lines above?

File: personalitytest/static/js/src/personalitytest-view.js (hunk 3 above)
Comment (fcheval) [inline personalitytest/static/js/src/personalitytest-view.js]: Done.

File: personalitytest/static/js/src/personalitytest-view.js [hunk 5]
```diff
@@ -97,16 +104,15 @@ function PersonalityTestXBlockStudent(runtime, element) {
...
+    $('.personality-test-form').on('click', '.save-button', function (e) {
+        e.preventDefault();
 
         var json = '[';
         var first = true;
         var errors = 0;
         $('select option:selected', element).each(function () {
             var that = $(this);
-            if (that.text() === '-') {
+            if (!that.value) {
```
Comment (tusbar) [inline personalitytest/static/js/src/personalitytest-view.js]: shouldn’t it be [`.va()`](http://api.jquery.com/val/) here?

Comment (fcheval) [inline personalitytest/static/js/src/personalitytest-view.js]: it doesn't work

File: personalitytest/static/js/src/personalitytest-view.js (hunk 4 above)
Comment (fcheval) [inline personalitytest/static/js/src/personalitytest-view.js]: Done

File: personalitytest/static/css/personalitytest.css (hunk 2 above)
Comment (fcheval) [inline personalitytest/static/css/personalitytest.css:12]: done

File: personalitytest/static/js/src/personalitytest-view.js (hunk 5 above)
Comment (tusbar) [inline personalitytest/static/js/src/personalitytest-view.js]: because, if you try and set a breakpoint, it will equal `false`.

Comment (fcheval) [inline personalitytest/static/js/src/personalitytest-view.js]: Done

File: personalitytest/static/js/src/personalitytest-view.js [hunk 6]
```diff
@@ -97,16 +103,15 @@ function PersonalityTestXBlockStudent(runtime, element) {
...
+        e.preventDefault();
 
         var json = '[';
         var first = true;
         var errors = 0;
         $('select option:selected', element).each(function () {
             var that = $(this);
-            if (that.text() === '-') {
+            if (that.val() === '') {
                 errors++;
```
Comment (tusbar) [inline personalitytest/static/js/src/personalitytest-view.js:188]: Also, there’s no need to loop through all the answers to just count the errors if just one is sufficient.

Break the loop if you find an error.

Comment (fcheval) [inline personalitytest/static/js/src/personalitytest-view.js:188]: done

**Pull request merged**